wallet/tickets.go

```diff
@@ -129,11 +131,68 @@ func (w *Wallet) TicketHashesForVotingAddress(votingAddr dcrutil.Address) ([]cha
 	return ticketHashes, err
 }
 
+// updateStakePoolInvalidTicket properly updates a previously marked Invalid pool ticket,
+// it then creates a new entry in the validly tracked pool ticket db.
+func (w *Wallet) updateStakePoolInvalidTicket(stakemgrNs walletdb.ReadWriteBucket, addrmgrNs walletdb.ReadBucket,
+	addr dcrutil.Address, ticket *chainhash.Hash, ticketHeight int64) error {
+	err := w.StakeMgr.RemoveStakePoolUserInvalTickets(stakemgrNs, addr, ticket)
+	if err != nil {
+		return err
+	}
+	poolTicket := &wstakemgr.PoolTicket{
+		Ticket:       *ticket,
+		HeightTicket: uint32(ticketHeight),
+		Status:       wstakemgr.TSImmatureOrLive,
+	}
+
+	err = w.StakeMgr.UpdateStakePoolUserTickets(stakemgrNs, addrmgrNs, addr, poolTicket)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
 // AddTicket adds a ticket transaction to the wallet.
 func (w *Wallet) AddTicket(ticket *dcrutil.Tx) error {
 	return walletdb.Update(w.db, func(tx walletdb.ReadWriteTx) error {
+		addrmgrNs := tx.ReadWriteBucket(waddrmgrNamespaceKey)
 		stakemgrNs := tx.ReadWriteBucket(wstakemgrNamespaceKey)
-		return w.StakeMgr.InsertSStx(stakemgrNs, ticket, w.VoteBits)
+
+		// Insert the ticket to be tracked and voted
+		err := w.StakeMgr.InsertSStx(stakemgrNs, ticket, w.VoteBits)
+		if err != nil {
+			return err
+		}
+
+		// Pluck the ticketaddress to indentify the stakepool user
```

Comment:
Should only do this extra logic for pool mode if pool mode is active, same as in chainntfns.go